polartools/diffraction.py

```diff
@@ -350,3 +350,274 @@ def fit_series(
+        passed.
+
+    Returns
+    -------
+    data : arrays with x, y and z information for 2D plot
+    """
+    nbp = 0
+    for series in range(1, len(scan_series), 3):
+        nbp = (
+            nbp
+            + int(scan_series[series] - scan_series[series - 1])
+            / scan_series[series + 1]
+            + 1
+        )
+
+    folder = kwargs.pop("folder", "")
+    if isinstance(source, (str, SpecDataFile)) and source != "csv":
+        if isinstance(source, str):
+            path = join(folder, source)
+            source = SpecDataFile(path)
+        if is_Bluesky_specfile(source):
+            _defaults = _bluesky_default_cols
+        else:
+            _defaults = _spec_default_cols
+
+    else:
+        _defaults = _bluesky_default_cols
+
+    if not positioner:
+        positioner = _defaults["positioner"]
+    if not detector:
+        detector = _defaults["detector"]
+    if not monitor:
+        monitor = _defaults["monitor"]
+    if len(scan_series) % 3:
+        raise ValueError(
+            f"expected 3*n={3*(len(scan_series)//3)} arguments, got {len(scan_series)}"
+        )
+
+    table = load_table(
+        scan_series[1],
+        source,
+        folder=folder,
+        **kwargs,
+    )
+    data_len = len(table[detector])
+    datax = [np.zeros(data_len) for i in range(int(nbp))]
+    datay = [np.zeros(data_len) for i in range(int(nbp))]
+    dataz = [np.zeros(data_len) for i in range(int(nbp))]
+
+    index = 0
+    for series in range(1, len(scan_series), 3):
+        start = scan_series[series - 1]
+        stop = scan_series[series]
+        step = scan_series[series + 1]
+        print("Intervals: {} to {} with step {}".format(start, stop, step))
+        for scan in range(start, stop + 1, step):
+            if var_series and var_series[0][0] == "#":
+                table = load_table(
+                    scan,
+                    source,
+                    folder=folder,
+                    **kwargs,
+                )
+                datay[index] = load_info(
+                    source, scan, info=var_series, folder=folder, **kwargs
+                )
+            elif var_series:
+                table = load_table(
+                    scan,
+                    source,
+                    folder=folder,
+                    **kwargs,
+                )
+                datay[index] = table[var_series]
+            else:
+                table = load_table(
+                    scan,
+                    source,
+                    folder=folder,
+                    **kwargs,
+                )
+                datay[index] = index
+
+            datax[index] = table[positioner]
+            if monitor:
+                dataz[index] = table[detector] / table[monitor]
+            else:
+                dataz[index] = table[detector]
+
+            if log:
+                for var in range(0, len(dataz[index])):
```

Comment:
Codacy complains about complexity when there are too many nested `for` or `if`, and I think the main issue is in this part. Would it work if lines 492 to 495 are replaced by this?

```python
dataz[index][dataz[index] == 0] = 1
dataz[index] = np.log10(dataz[index])
```

Reply:
Great. Looks like that solved it. Mr. Codacy is happy.